refinery_core/src/drivers/ducdb.rs

```diff
@@ -0,0 +1,56 @@
+use crate::traits::sync::{Migrate, Query, Transaction};
+use crate::Migration;
+use duckdb::{Connection as DqlConnection, Error as DqlError};
+use time::format_description::well_known::Rfc3339;
+use time::OffsetDateTime;
+
+fn query_applied_migrations(
+    transaction: &DqlConnection,
+    query: &str,
+) -> Result<Vec<Migration>, DqlError> {
+    let mut stmt = transaction.prepare(query)?;
+    let mut rows = stmt.query([])?;
+    let mut applied = Vec::new();
+    while let Some(row) = rows.next()? {
+        let version = row.get(0)?;
+        let applied_on: String = row.get(2)?;
+        // Safe to call unwrap, as we stored it in RFC3339 format on the database
```

Comment:
Thanks for the useful comment! Perhaps we could use `expect()` to add a bit of extra context? Could be useful in the odd situation where this actually panics (bitflips, data corruption, etc)